### PR TITLE
Add "--down" flag that will rollback previous migration via code rather than from DB.

### PR DESCRIPTION
**Pull request by incorvia**
I've created this PR as a good faith effort to add another option to give developers an alternative work flow for editing migrations. By running down migrations from code, it gives the developer flexibility to quickly fix 'down' migrations that throw exceptions but are committed to the database. I understand this likely won't be merged as it doesn't match the philosophy of the maintainer is trying to achieve with this project. Even still, I thought since I did the work I'd give the maintainers the option to merge, suggest improvements if they'd like, or simply close.

If nothing else I appreciate the opportunity to review the existing code and have learned a lot as I've only been developing in Python for about 5 days now.

### CODE REVIEW

File: migrate_anything/migrator.py [hunk 1]
```diff
@@ -147,7 +147,41 @@ def _apply_migrations(migrations):
...
+    except IndexError:
+        logger.info("No migrations to rollback found.")
+        return
+
+    # Check to make sure the migration exists
+    migration_exists = last_run_migration in current_keys
+
+    if migration_exists:
+        module = migrations[last_run_migration]
+        module.down()
```
Comment (lietu) [inline migrate_anything/migrator.py:178]: The issue here is that what normally happens is that the `down` is run as stored in the DB, and in this one it's run from the current state of the file. The `--down` argument does not make this huge distinction clear.

**Review by lietu — CHANGES_REQUESTED** [review]
Please update `README.rst` to document usage clearly, and propose clearer naming for `--down`.

**Comment by lietu** [comment]
I will also work on the main repository to update it to modern build infrastructure and Python versions, Python 2.7 is long dead so dropping support, switching to GitHub actions, documenting development on the tool itself a bit better, updating dev requirements. Stuff like that. I'll try to finish this soon, you will likely have to merge those changes to your branch afterwards.

File: migrate_anything/migrator.py (hunk 1 above)
Comment (lietu) [reply migrate_anything/migrator.py:178]: Additionally `--down` does not make it clear it's only going to run "down" on the latest, so something more like `--down-one` would be more descriptive

Comment (lietu) [reply migrate_anything/migrator.py:178]: Or `--down-latest`

**Comment by lietu** [comment]
The repo seems to be in a more reasonable state now, please sync your fork with this and check the README for updated instructions on e.g. `pre-commit` setup.

**Comment by incorvia** [comment]
@lietu I agree the name leaves much to be desired. Some options here:

- We introduce two flags "--revert 
" which indicates a number to revert and "-m or --mode" with valid values of "db" or "module" (or "code" if you'd prefer) which indicates whether the reversions should happen from the db or from the code.
- We simply rename the flag. Some options are "--revert-one", "--down-one" and docs make it clear it's from code and the purpose is to rescue from bad downs values in DB. 
- We rename option to --revert-from-code and the value is a number to revert. 
- Some other suggestion you might have.

**Comment by lietu** [comment]
Quick thoughts

1) You primarily need this feature when developing a migration, and you should really only work on one at a time, so that should be the default assumption
2) You can run the command multiple times if you really need to
3) If the name is `revert` instead of `down` the distinction of what is run from DB and what from file should be more obvious

So I'd prefer just `--revert-latest` / `--revert-latest-from-code` / `--revert-one` / `--revert-one-from-code`. I think the `from-code` can be documented in the `ArgumentParser` + `README` well enough that it's not necessary.

File: migrate_anything/migrator.py [hunk 2]
```diff
@@ -147,7 +147,41 @@ def _apply_migrations(migrations):
...
+        last_run_migration = applied_keys[-1]
+    except IndexError:
+        logger.info("No migrations to rollback found.")
+        return
+
+    # Check to make sure the migration exists
+    migration_exists = last_run_migration in current_keys
+
+    if migration_exists:
+        module = migrations[last_run_migration]
```
Comment (lietu) [inline migrate_anything/migrator.py:177]: Please include a message as in the other case:

```
logger.info("Undoing migration {}".format(last_run_migration))
```

**Comment by incorvia** [comment]
@lietu Changes I believe I've incorporated all your suggestions into the PR.

**Review by lietu — APPROVED** [review]

**Comment by lietu** [comment]
Resolves #3

**Comment by lietu** [comment]
Released as version `0.1.8` https://pypi.org/project/migrate-anything/0.1.8/

**Comment by lietu** [comment]
Admittedly I probably should've bumped 0.1.7 to 0.2.0 at minimum considering it breaks compatibility with Python `<3.6`

**Pull request merged**